docs-ref-services/latest/msal-python-overview.md

```diff
@@ -0,0 +1,106 @@
+
+[Get started](/azure/active-directory/develop/quickstart-v2-python-webapp) | [API reference Documentation](https://github.com/AzureAD/microsoft-authentication-library-for-python/wiki) (GitHub) | [Samples](https://aka.ms/aaddevsamplesv2) |
+
+
+## Prerequisites
+
+- An Azure account with an active subscription. [Create a free account][azure_sub].
+- [Python 3.6+](https://www.python.org/downloads/)
+
+## Install the package
+
+Install the MSAL for Python package. You can find MSAL Python on [Pypi](https://pypi.org/project/msal/).
+```Bash
+pip install msal
+```
+
+## Setting up
+
+Before using MSAL Python [register your application](/azure/active-directory/develop/quickstart-v2-register-an-app) with the Microsoft identity platform.
+
+## Usage
+
+Acquiring tokens with MSAL Python follows this 3-step pattern. This is the high level conceptual pattern. There will be some variations for different flows. They are demonstrated in [runnable samples hosted right in this repo](https://github.com/AzureAD/microsoft-authentication-library-for-python/tree/dev/sample).
+
+1. MSAL proposes a clean separation between [public client applications, and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). Therefore, create either a `PublicClientApplication` or a `ConfidentialClientApplication` instance, and reuse it during the lifecycle of your app. The following example shows a `PublicClientApplication`:
+
+   ```python
+   from msal import PublicClientApplication
+   app = PublicClientApplication(
+       "your_client_id",
+       authority="https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
+   ```
+
+   Later, each time you would want an access token, you start by:
+   ```python
+   result = None  # It is just an initial value. Please follow instructions below.
+   ```
+
+2. The API model in MSAL provides you explicit control on how to utilize token cache. This cache part is technically optional, but we highly recommend you to harness the power of MSAL cache. It will automatically handle the token refresh for you.
+
+   ```python
+   # We now check the cache to see
+   # whether we already have some accounts that the end user already used to sign in before.
+   accounts = app.get_accounts()
+   if accounts:
+       # If so, you could then somehow display these accounts and let end user choose
+       print("Pick the account you want to use to proceed:")
+       for a in accounts:
+           print(a["username"])
+       # Assuming the end user chose this one
+       chosen = accounts[0]
+       # Now let's try to find a token in cache for this account
+       result = app.acquire_token_silent(["your_scope"], account=chosen)
+   ```
+
+3. Either there is no suitable token in the cache, or you chose to skip the previous step. Send a request to Azure AD to obtain a token. There are different methods based on your client type and scenario. Here we demonstrate a placeholder flow.
```

Comment:
@mmacy Good question. Please help us decide a better way to convey the following intention in doc and in code.

> There are a bunch of `acquire_token_xyz(...)` methods in MSAL Python's API (see the column at the right hand side [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal#api-changes)). Here in our "3-steps pattern", step 2 uses `acquire_token_silent()` which attempts to utilize tokens inside cache.
>
> If there is no suitable token in the cache or you've chosen to skip the previous step, step 3 would need to send a request to Azure AD to get a token, by using ANY of the remaining `acquire_token_xyz(...)` methods, and their actual usage are demonstrated in [their own samples](https://github.com/AzureAD/microsoft-authentication-library-for-python/tree/dev/sample).

Of course, we could use one specific method to complete this paragraph and the following code snippet, but then it would arguably lose the intention to say "other acquire_token_by_xyz(...) methods are equally applicable here".

What do you think?